torch/_dynamo/funcname_cache.py

```diff
@@ -0,0 +1,53 @@
+import tokenize
+
+cache = {}
+
+
+def clearcache():
+    cache.clear()
+
+
+def _add_file(filename):
+    try:
+        with open(filename) as f:
+            tokens = list(tokenize.generate_tokens(f.readline))
+    except Exception:
+        cache[filename] = {}
+        return
+
+    # NOTE: undefined behavior if file is not valid Python source,
+    # since tokenize will have undefined behavior.
+    result = {}
+    cur_name = ""
+    cur_indent = 0
+    significant_indents = []
+
+    for i, token in enumerate(tokens):
+        if token.type == tokenize.INDENT:
+            cur_indent += 1
+        elif token.type == tokenize.DEDENT:
+            cur_indent -= 1
+            # possible end of function or class
+            if significant_indents and cur_indent == significant_indents[-1]:
+                significant_indents.pop()
+                cur_name = cur_name.rpartition(".")[0]
```

Comment:
why not just track cur_name as a list so you can pop/push, and join it when you want to write it to result? (You could cache the join if you want to avoid repeatedly joining)

Reply:
The repeated joins are what I'm really trying to avoid here.